src/Component/RandomDogs.tsx

```diff
@@ -0,0 +1,30 @@
+import {FunctionComponent, useCallback, useEffect,useState} from 'react'
+import * as helpers from '../Helpers'
+
+interface IDogsProps {
+    urlForAPICall: string
+}
+
+const RandomDogs: FunctionComponent<IDogsProps> = (props: IDogsProps) => {
+    const [dogs,setDogs] = useState<string[]>([]);
+    const [compIsMounted,setCompIsMounted] = useState<boolean>(false); // Mounts once the fetchDogs function
+    const fetchDogs = useCallback(()=> helpers.retrieveData(props.urlForAPICall,dogs,setDogs),[props.urlForAPICall,dogs])

```

Comment:
How come you decided to use useCallback here?

Reply:
I used callback because although the function was retrieving data correctly,
the component was not re-rendering to make the changes visible to the user.

By searching, I found the useCallback hook that addressed elegantly this problem.